plugins/woocommerce-docs/tests/src/Manifest/ManifestProcessorTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace WooCommerceDocs\Tests\Manifest;
+
+use WooCommerceDocs\Manifest\ManifestProcessor;
+use WP_UnitTestCase;
+use WooCommerceDocs\Data\DocsStore;
+
+/**
+ * Class ManifestProcessorTest
+ *
+ * @package WooCommerceDocs\Tests\Manifest
+ */
+class ManifestProcessorTest extends WP_UnitTestCase {
+
+	/**
+	 * Test processing a manifest into WordPress posts and categories.
+	 */
+	public function test_process_manifest() {
+
+		$manifest = json_decode( file_get_contents( __DIR__ . '/fixtures/manifest.json' ), true );
+		$md_file  = file_get_contents( __DIR__ . '/fixtures/test.md' );
+
+		// Mock the wp_remote_get function with a filter.
+		add_filter(
+			'pre_http_request',
+			function ( $preempt, $args, $url ) use ( $md_file ) {
+				return array(
+					'response' => array(
+						'code'    => 200,
+						'message' => 'OK',
+					),
+					'body'     => $md_file,
+				);
+			},
+			10,
+			3
+		);
+
+		ManifestProcessor::process_manifest( $manifest, 123 );
```

Comment:
This makes me wonder whether `process_manifest` is maybe doing too many things ... it processes the manifest _and_ calls the WordPress API to insert the results of the processing. Ideally the test would just give it an input and look at the output, but here we check the output _and_ whether it was put into the correct place (the WordPress database). 

No action here but just a thought. I think this is what people refer to when saying that writing tests can improve the architecture. 🤔

Reply:
Youre right. I was concretely aware of this when developing but held off refactoring for now (i even started writing the classes) in favor of getting some basic features done. It wont be a big effort to extract this to separate classes in upcoming PRs.

I can do that as a follow up once this stack of feature PRs is merged :) thanks for this kind of high quality feedback btw. Its exactly the kind of stuff we should be tackling to keep quality in check.

Reply:
Great, looking forward! ☺️